### ISSUE TITLE
some question about paper

### ISSUE THREAD

**Issue by helloyuning** [chensong1995/HybridPose#74]
Hi, thanks for your grate work and sharing

I have a question about symmetrical correspondences. 
Specifically how are symmetric labels made, and what does the network use to determine if two pixels are symmetric or not？

Thanks

**Comment by chensong1995** [comment]
Hi helloyuning,

Thanks for your interest in our work!

- The symmetric labels are generated by [SymSeg](https://github.com/aecins/symseg). Please refer to #13 for related discussions.
- The ResNet predicts [a segmentation mask](https://github.com/chensong1995/HybridPose/blob/e9a4bb6115a9d3609cdf9b48691528aea3e17f20/trainers/coretrainer.py#L85). For each pixel within the foreground region defined by the mask prediction, the network also predicts a symmetric counterpart for the pixel.

I hope this helps! Let me know if you have further questions.

**Comment by helloyuning** [comment]
> Hi helloyuning,
> 
> Thanks for your interest in our work!
> 
> * The symmetric labels are generated by [SymSeg](https://github.com/aecins/symseg). Please refer to [Question: Difficulties when creating the symmetries using SymSeg #13](https://github.com/chensong1995/HybridPose/issues/13) for related discussions.
> * The ResNet predicts [a segmentation mask](https://github.com/chensong1995/HybridPose/blob/e9a4bb6115a9d3609cdf9b48691528aea3e17f20/trainers/coretrainer.py#L85). For each pixel within the foreground region defined by the mask prediction, the network also predicts a symmetric counterpart for the pixel.
> 
> I hope this helps! Let me know if you have further questions.

Thanks for your reply

I am confused about the training.
1. 
2. 

> Hi helloyuning,
> 
> Thanks for your interest in our work!
> 
> * The symmetric labels are generated by [SymSeg](https://github.com/aecins/symseg). Please refer to [Question: Difficulties when creating the symmetries using SymSeg #13](https://github.com/chensong1995/HybridPose/issues/13) for related discussions.
> * The ResNet predicts [a segmentation mask](https://github.com/chensong1995/HybridPose/blob/e9a4bb6115a9d3609cdf9b48691528aea3e17f20/trainers/coretrainer.py#L85). For each pixel within the foreground region defined by the mask prediction, the network also predicts a symmetric counterpart for the pixel.
> 
> I hope this helps! Let me know if you have further questions.

Thanks for your reply.
I am confused about the training stage.
1. Are the keypoints output in the prediction stage voted for, or are they keypoints output end-to-end
2. Do the initial pose and refienment pose require neural network training, and how are their loss functions constituted

Best

**Comment by chensong1995** [comment]
Hi helloyuning,

Thanks for the follow-up questions!

1. HybridPose does not perform keypoint voting at training time. The training supervision is the voting vector instead of the result of the voting.
2. HybridPose does not estimate the pose parameters either at training time. The training supervision is applied on the intermediate representations (keypoints, edge vectors, symmetry correspondences).

I hope this helps! Let me know if you have further concerns.
